### PR TITLE
fix(levm): handle type conversions

### LINKED ISSUE #1074: LEVM: add `#[forbid(clippy::unwrap_used)]`, `#[forbid(clippy::expect_used)]`, and `#[forbid(clippy::panic)]`
1. Add the lints `#[forbid(clippy::unwrap_used)]`, `#[forbid(clippy::expect_used)]`, and `#[forbid(clippy::panic)]` to the `levm` module.
2. Handle the errors or propagate them as needed.

### PR DESCRIPTION
**Pull request by JereSalo**
**Motivation**

**Description**

Partially Closes #1074

### CODE REVIEW

File: crates/vm/levm/clippy_delete.txt
Comment (ilitteri) [inline crates/vm/levm/clippy_delete.txt:1]: Remove this file.

**Review by ilitteri — CHANGES_REQUESTED** [review]
The PR looks good, there are a few tweaks to do though. Left some comments. Let's do a similar approach as suggested in https://github.com/lambdaclass/lambda_ethereum_rust/pull/1107#pullrequestreview-2427589737 for the `ConversionError`

File: crates/vm/levm/src/opcodes.rs [hunk 1]
```diff
@@ -333,3 +333,15 @@ impl TryFrom<u8> for Opcode {
...
+    #[allow(clippy::as_conversions)]
+    pub fn to_usize(self) -> usize {
+        self as usize
+    }
+
+    #[allow(clippy::as_conversions)]
+    pub fn to_u8(self) -> u8 {
+        self as u8
+    }
+}
```
[the remark this answers is not in the document]
Comment (JereSalo) [reply crates/vm/levm/src/opcodes.rs]: What I actually need is to convert he Opcode into an u8 or usize.
I can replace it for

```rust
impl From<Opcode> for u8 {
    fn from(opcode: Opcode) -> Self {
        opcode as u8
    }
}

impl From<Opcode> for usize {
    fn from(opcode: Opcode) -> Self {
        opcode as usize
    }
}
```

File: crates/vm/levm/clippy_delete.txt
Comment (JereSalo) [reply crates/vm/levm/clippy_delete.txt:1]: Done!

File: crates/vm/levm/src/constants.rs [hunk 2]
```diff
@@ -1,5 +1,7 @@
 use ethereum_rust_core::U256;
 
+use crate::errors::{InternalError, VMError};
```
Comment (ilitteri) [inline crates/vm/levm/src/constants.rs:3]: Let's try to keep the imports together.
```suggestion
use crate::errors::{InternalError, VMError};
```

File: crates/vm/levm/src/vm.rs [hunk 3]
```diff
@@ -441,13 +441,17 @@ impl VM {
...
+                .try_into()
+                .map_err(|_| VMError::Internal(InternalError::ConversionError))?;
+            let mut creation_cost = 200 * code_length;
             creation_cost += 32000;
             report.add_gas_with_max(creation_cost, max_gas);
             // Charge 22100 gas for each storage variable set
 
             // GInitCodeword * number_of_words rounded up. GinitCodeWord = 2
-            let number_of_words = initial_call_frame.calldata.chunks(32).len() as u64;
+            let number_of_words: u64 = initial_call_frame.calldata.chunks(32).len().try_into().map_err(|_| VMError::Internal(InternalError::ConversionError))?;
```
Comment (ilitteri) [inline crates/vm/levm/src/vm.rs:454]:
```suggestion
            let number_of_words: u64 = initial_call_frame.calldata.chunks(WORD_SIZE).len().try_into().map_err(|_| VMError::Internal(InternalError::ConversionError))?;
```

**Review by ilitteri — APPROVED** [review]
Amazing work! LGTM. Left two minor comments.

File: crates/vm/levm/src/vm.rs [hunk 4]
```diff
@@ -638,17 +648,24 @@ impl VM {
...
+                    sender_address.as_bytes(),
+                    &salt_bytes,
+                    init_code_hash.as_bytes(),
+                ]
+                .concat(),
+            )
+            .as_bytes()
+            .get(12..)
+            .expect("Failed to get create2 address"),
+        )
```
Comment (ilitteri) [inline crates/vm/levm/src/vm.rs:668]: I think these changes don't belong to this PR.

File: crates/vm/levm/src/vm.rs [hunk 5]
```diff
@@ -772,9 +789,7 @@ impl VM {
         );
 
         let new_address = match salt {
-            Some(salt) => {
-                Self::calculate_create2_address(current_call_frame.msg_sender, &code, salt)
-            }
+            Some(salt) => Self::calculate_create2_address(current_call_frame.to, &code, salt),
```
Comment (ilitteri) [inline crates/vm/levm/src/vm.rs:792]: Ditto. By the way, this is what is making the tests fail.

File: crates/vm/levm/src/vm.rs (hunk 4 above)
Comment (JereSalo) [reply crates/vm/levm/src/vm.rs:668]: I didn't realize those changes were reverted in levm_fixes, I will change it.

File: crates/vm/levm/src/vm.rs (hunk 5 above)
Comment (JereSalo) [reply crates/vm/levm/src/vm.rs]: Changed that [here](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1110/commits/c4f50b8540a7d2970900538e6b26074c8fa26a4c)

**Review by ilitteri — APPROVED** [review]
